kubernetes.el

```diff
@@ -38,6 +38,14 @@
 (require 'kubernetes-overview)
 (require 'kubernetes-el-tramp)
 
+(defmacro kubernetes--with-proxy (&rest body)
```

Comment:
Given that the process ledger doesn't have a TTL mechanism for the proxy process, let's compromise and give the `--with-proxy` macro the option to explicitly request to terminate the proxy after it's done. 
```emacs-lisp
(defmacro kubernetes--with-proxy (&optional terminate-on-done &rest body)
  ...)
```

This would require `get-proxy-process` to return a sentinel value denoting whether a new proxy process was created, or if it's simply returning an existing one. If the latter, then `terminate-on-done` shall be a no-op.

Naturally, this will also require a `release-proxy-process` method on the process ledger. This can be a no-op if there's no existing proxy process.

We might also consider having `--with-proxy` kick off a timer, after which `release-proxy-process` gets called regardless.

Reply:
I mostly just think it'd be a terrible idea to have a live proxy running hidden in the background indefinitely.